webrender/src/tiling.rs

```diff
@@ -714,28 +730,17 @@ impl AlphaBatcher {
                         let group = &ctx.clip_scroll_group_store[clip_scroll_group_index.0];
                         let transform_kind = group.xf_rect.as_ref().unwrap().kind;
                         let prim_metadata = ctx.prim_store.get_metadata(prim_index);
-                        let needs_clipping = prim_metadata.clip_task.is_some();
-                        let needs_blending = transform_kind == TransformedRectKind::Complex ||
-                                             !prim_metadata.is_opaque ||
-                                             needs_clipping;
-                        let blend_mode = ctx.prim_store.get_blend_mode(needs_blending, prim_metadata);
-                        let needs_clipping_flag = if needs_clipping {
-                            NEEDS_CLIPPING
-                        } else {
-                            AlphaBatchKeyFlags::empty()
-                        };
-                        let flags = match transform_kind {
-                            TransformedRectKind::AxisAligned => AXIS_ALIGNED | needs_clipping_flag,
-                            _ => needs_clipping_flag,
-                        };
+                        debug_assert!(!prim_metadata.needs_clipping());
```

Comment:
I'd rather not tie up clipping, axis-alignment, and opaque-ness properties together just yet.